arch/src/x86_64/mod.rs

```diff
@@ -1010,6 +1028,19 @@ pub fn initramfs_load_addr(
     Ok(aligned_addr)
 }
 
+/// Returns the host cache info.
+///
+/// Return None if CPU does not support.
+fn get_host_cpu_cache_info(func: u32) -> Option<std::arch::x86_64::CpuidResult> {
+    // This function is the same as the x86_cpu_get_cache_cpuid of QEMU, but 0x8000001d
+    // is used for all AMD cache information leaves. So it is not support.
+    let leaf = unsafe { std::arch::x86_64::__cpuid(func & 0xffff_fff0) };
```

Comment:
This is just confusing vs writing `0x8000_0000`.

Reply:
I think it may be necessary to support the case of 0x4 and 0x2 :) Although it is not used here.

```
get_host_cpu_cache_info(0x2)
get_host_cpu_cache_info(0x8000_0006)
```

Reply:
You haven't added anything for any other leaves so you should do it as I proposed without this function and checking for the maximum leaf where you need to.